pubnub/event_engine/models/states.py

```diff
@@ -180,15 +182,15 @@ def __init__(self, context: PNContext) -> None:
     def on_enter(self, context: Union[None, PNContext]):
         self._context.update(context)
         super().on_enter(self._context)
-        return effects.HandshakeReconnectEffect(self._context.channels,
-                                                self._context.groups,
-                                                attempts=self._context.attempt,
-                                                reason=self._context.reason,
-                                                timetoken=self._context.timetoken)
+        return invocations.HandshakeReconnectInvocation(self._context.channels,
+                                                        self._context.groups,
+                                                        attempts=self._context.attempt,
+                                                        reason=self._context.reason,
+                                                        timetoken=self._context.timetoken)
 
     def on_exit(self):
         super().on_exit()
-        return effects.CancelHandshakeReconnectEffect()
+        return invocations.CancelHandshakeReconnectInvocation()
 
     def disconnect(self, event: events.DisconnectEvent, context: PNContext) -> PNTransition:
```

Comment:
IMHO all transitions should be part of `Transition function` that was mentioned in the State Machine Architecture decision. This shouldn't be the logic of any State. Let's keep this comment so that others are aware of this difference.